src/command.mli

```diff
@@ -0,0 +1,79 @@
+open! Stdune
+(** This module replaces arg_spec.mli in the post-arrow Dune. The main
+    change is the constructor [Dyn] which now takes [static t Build.s]
+    instead of ['a -> (unit, static) t], which allows us to get rid of
+    the first type parameter ['a] in [('a, 'b) Arg_spec.t]. *)
+
+(** Command line arguments specification *)
+
+(** This module implements a small DSL to specify the command line
+    argument of a program as well as the dependencies and targets of
+    the program at the same time.
+
+    For instance to represent the argument of [ocamlc -o src/foo.exe
+    src/foo.ml], one might write:
+
+    {[
+      [ A "-o"
+      ; Target (Path.relatie  dir "foo.exe")
+      ; Dep    (Path.relative dir "foo.ml")
+      ]
+    ]}
+
+    This DSL was inspired from the ocamlbuild API.  *)
+
+open! Import
+
+(** [A] stands for "atom", it is for command line arguments that are
+    neither dependencies nor targets.
+
+    [Path] is similar to [A] in the sense that it defines a command
+    line argument that is neither a dependency or target. However, the
+    difference between the two is that [A s] produces exactly the
+    argument [s], while [Path p] produces a string that depends on
+    where the command is executed. For instance [Path (Path.of_string
+    "src/foo.ml")] will translate to "../src/foo.ml" if the command is
+    started from the "test" directory.  *)
+
+type static = Static
+type dynamic = Dynamic
+
+type _ t =
+  | A        : string -> _ t
+  | As       : string list -> _ t
+  | S        : 'a t list -> 'a t
+  | Concat   : string * 'a t list  -> 'a t
+  | Dep      : Path.t -> _ t
+  | Deps     : Path.t list -> _ t
+  | Target   : Path.t -> dynamic t
+  | Path     : Path.t -> _ t
+  | Paths    : Path.t list -> _ t
+  | Hidden_deps    : Dep.Set.t -> _ t
+  | Hidden_targets : Path.t list -> dynamic t
+  | Dyn      : static t Build.s -> dynamic t
+  | Fail     : fail -> _ t
+
+(* Create a dynamic command line arguments. *)
+val dyn_args : string list Build.s -> dynamic t
+
+(* Create a dynamic dependencies. *)
+val dyn_dep  : Path.t Build.s -> dynamic t
+val dyn_deps : Path.t list Build.s -> dynamic t
```

Comment:
To be consistent with similar functions in other modules, these function should take `Dep.t` values rather than `Path.t` ones. Alternatively they could be renamed. Although, given that each of these is used only once in the rest of the code base, I would just inline them.

Reply:
These can't take an arbitrary `Dep.t` here because we're adding the path both as a dependency and as an argument on the command line. This confusion seems very hard to avoid in this module in general.

I'm OK with inlining though.

Reply:
I'm fine with inlining too. Will do that.

Reply:
Done.